diff -ru zfs-2.1.15.orig/etc/systemd/system/zfs-import-cache.service.in zfs-2.1.15/etc/systemd/system/zfs-import-cache.service.in
--- zfs-2.1.15.orig/etc/systemd/system/zfs-import-cache.service.in	2024-02-27 00:04:26.084684081 +0100
+++ zfs-2.1.15/etc/systemd/system/zfs-import-cache.service.in	2024-05-03 16:15:44.434000000 +0200
@@ -5,7 +5,8 @@
 Requires=systemd-udev-settle.service
 After=systemd-udev-settle.service
 After=cryptsetup.target
-After=multipathd.service
+#XCP-ng: break dependency loop, no ZFS over mpath?
+#After=multipathd.service
 After=systemd-remount-fs.service
 Before=zfs-import.target
 ConditionFileNotEmpty=@sysconfdir@/zfs/zpool.cache
